src/ai/backend/manager/models/vfolder.py

```diff
@@ -530,13 +526,24 @@ async def prepare_vfolder_mounts(
         extra_vf_conds=extra_vf_conds,
     )
 
+    # Fast-path for empty requested mounts
+    if not accessible_vfolders:
+        return []
```

Comment:
I think we should handle this case as well.
Instead of directly applying my suggestion, could we simplify the matching?
```suggestion
    if not accessible_vfolders:
        if requested_vfolder_names:
            raise VFolderNotFound(f"VFolder {requested_vfolder_names[0]} is not found or accessible.")
        else:
            return []
```